torch/_decomp/decompositions.py

```diff
@@ -728,23 +728,37 @@ def slice_forward(
     start_val = start if start is not None else 0
     end_val = end if end is not None else sys.maxsize  # 2^63 - 1
 
-    if guard_size_oblivious(start_val < 0):
+    # negative indexing case
```

Comment:
we need more testing for when end is None. 
for example if end is None and start is unabcked what shall we do?
 we can assume sys.maxsize<start and check on it we should not always fail!